superdesk/storage/async_storage/media_storage.py

```diff
@@ -0,0 +1,265 @@
+# -*- coding: utf-8; -*-
+#
+# This file is part of Superdesk.
+#
+# Copyright 2024 to present Sourcefabric z.u. and contributors.
+#
+# For the full copyright and license information, please see the
+# AUTHORS and LICENSE files distributed with this source code, or
+# at https://www.sourcefabric.org/superdesk/license
+
+import os
+import json
+import gridfs
+import logging
+
+from bson import ObjectId
+from motor.motor_asyncio import AsyncIOMotorGridFSBucket, AsyncIOMotorGridOut
+from typing import Any, BinaryIO, Dict, MutableMapping, Optional, Union, List, cast
+
+from eve.io.media import MediaStorage
+from superdesk.core.app import SuperdeskAsyncApp
+from superdesk.storage.desk_media_storage import format_id
+from superdesk.storage.mimetype_mixin import MimetypeMixin
+
+logger = logging.getLogger(__name__)
+
+
+class GridFSMediaStorageAsync(MediaStorage, MimetypeMixin):
+    """
+    The GridFSMediaStorageAsync class stores files into GridFS
+    using asynchronous approach.
+
+    .. versionadded:: 3.0
+    """
+
+    _fs: Dict[str, AsyncIOMotorGridFSBucket]
+
+    app: SuperdeskAsyncApp
+
+    def __init__(self, app: SuperdeskAsyncApp):
+        self.app = app
```

Comment:
It would be good to not rely on any app specifics, making this storage interface generic, only relying on the PyMongo library. We can then add async GridFS functionality into the existing async mongo module (under `superdesk.core.mongo`).

Reply:
done.